libs/model/src/thread/CreateThreadReaction.command.ts

```diff
@@ -0,0 +1,84 @@
+import { InvalidState, type Command } from '@hicommonwealth/core';
+import * as schemas from '@hicommonwealth/schemas';
+import { models } from '../database';
+import { isCommunityAdminOrTopicMember } from '../middleware';
+import { verifyReactionSignature } from '../middleware/canvas';
+import { mustExist } from '../middleware/guards';
+import { getVotingWeight } from '../services/stakeHelper';
+
+export const CreateThreadReactionErrors = {
+  ThreadArchived: 'Thread is archived',
+};
+
+export function CreateThreadReaction(): Command<
+  typeof schemas.CreateThreadReaction
+> {
+  return {
+    ...schemas.CreateThreadReaction,
+    auth: [
+      isCommunityAdminOrTopicMember(
+        schemas.PermissionEnum.CREATE_THREAD_REACTION,
+      ),
+      verifyReactionSignature,
+    ],
+    body: async ({ actor, payload }) => {
+      const thread = await models.Thread.findOne({
+        where: { id: payload.thread_id },
+      });
+      if (!mustExist('Thread', thread)) return;
+      if (thread.archived_at)
+        throw new InvalidState(CreateThreadReactionErrors.ThreadArchived);
+
+      const address = await models.Address.findOne({
```

Comment:
This address check on the community with the userId passed in the JWT seems like a pretty common thing we do, shouldn't it be abstracted into a middleware function?

Reply:
yep, I'm building the pattern inside authorization middleware... will need refactoring soon after we finish the first batch of commands ... checkout the next command in the pipeline for more changes around this area